fetch_response.go

```diff
@@ -355,10 +357,13 @@ func encodeKV(key, value Encoder) ([]byte, []byte) {
 	return kb, vb
 }
 
-func (r *FetchResponse) AddMessage(topic string, partition int32, key, value Encoder, offset int64) {
+func (r *FetchResponse) AddMessageWithTimestamp(topic string, partition int32, key, value Encoder, offset int64, timestamp time.Time, version int8) {
```

Comment:
This and below are exported function and this change is going to be a breaking change. Let's see how we can handle them. @bai do you have an idea?

Reply:
You'll need to keep both exported functions, and make one delegates to the other!

Reply:
@varun06 , I think I did preserve the original API call reimplementing the original function using this one (https://github.com/Shopify/sarama/pull/1258/files/e1eda41894f763094e70b74f6dbf958c4cbbd315#diff-be942b2617e64413fb251f24e4bd4ea2R388). Same for `AddRecord`.

Am I missing something ?

Reply:
The “AddMessage” method is still there (check the link I provided above). Also note that there are multiple tests that would have got broken otherwise.

Reply:
Apologies, Do we have AddRecord method also?

Reply:
It looks like @zimin2000 preserved the original `AddRecord` and `AddMessage` names & signatures below starting on line 388. Perhaps they should each be moved right above their respective `WithTimestamp` versions.

Reply:
Yes, “AddRecord” also kept.

Reply:
Cool Bean. Let's get it going then. @bai can we merge this please?